projects/rocprofiler-systems/source/lib/rocprof-sys/library/rocprofiler-sdk.cpp

```diff
@@ -906,16 +990,224 @@ get_kernel_dispatch_timestamps()
     return _v;
 }
 
+#if(ROCPROFILER_VERSION >= 600)
+// To handle events without finalization, perfetto push must occur in start
+// Allows capture of worker thread implicit and sync tasks
+void
+ompt_tracing_callback_start(rocprofiler_callback_tracing_record_t record,
+                            rocprofiler_user_data_t* /*user_data*/,
+                            rocprofiler_timestamp_t ts)
+{
+    std::string_view _name;
+#    if(ROCPROFSYS_OMPT_EXPERIMENTAL_TYPE_NAMES > 0)
+    std::string check_name;
+    if(record.kind == ROCPROFILER_CALLBACK_TRACING_OMPT)
+    {
+        check_name = ompt_get_detailed_callback_name(record);
+        _name      = check_name;
+    }
+    if(_name.empty())
+        _name = tool_data->callback_tracing_info.at(record.kind, record.operation);
+#    else
+    _name = tool_data->callback_tracing_info.at(record.kind, record.operation);
+#    endif
+
+#    if(!(ROCPROFSYS_OMPT_EXPERIMENTAL_TYPE_NAMES > 0))
+    // Forces omp_parallel begin and end to have same name, allowing perfetto track to
+    // connect. This will be changed in the future
+    if(record.operation == ROCPROFILER_OMPT_ID_parallel_begin) _name = "omp_parallel";
+#    endif
```

Comment:
```suggestion
    std::string_view _name = tool_data->callback_tracing_info.at(record.kind, record.operation);
```
Let's keep these "ROCPROFSYS_OMPT_EXPERIMENTAL_TYPE_NAMES" changes on a feature branch, for now.

Reply:
The renaming of a record with operation `ROCPROFILER_OMPT_ID_parallel_begin` must remain for the tracks to connect properly